language/tools/move-cli/src/main.rs

```diff
@@ -336,10 +350,33 @@ fn publish(args: &Move, republish: bool, files: &[String]) -> Result<OnDiskState
             CompiledUnit::Module { module, .. } => modules.push(module),
         }
     }
-    Ok(OnDiskStateView::create(
-        storage_dir.to_path_buf(),
-        &modules,
-    )?)
+
+    if !ignore_breaking_changes {
+        let view = OnDiskStateView::create(storage_dir.to_path_buf(), &[])?;
+        for m in &modules {
+            let id = m.self_id();
+            if let Ok(old_m) = view.get_compiled_module(&m.self_id()) {
+                let old_api = Module::new(&old_m);
+                let new_api = Module::new(m);
+                let compat = Compatibility::check(&old_api, &new_api);
+                if !compat.is_fully_compatible() {
+                    println!("Breaking change detected--publishing aborted. Re-run with --ignore-breaking-changes to publish anyway.")
```

Comment:
Possibly a larger question for the CLI as well, but I'm thinking this should probably be an `eprintln!` instead since this is a warning/error.

Reply:
Yeah, good call. I'll do that for now, and in the future I'd like to get rid of all `print`'s and turn them into writes to a buffer. I think we need to do this in order to run CLI tests concurrently in the same process.